Src/WindowsServer/WindowsServer.Shared/ConfigurationTelemetryInitializer.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.ApplicationInsights.WindowsServer.Implementation;
+
+namespace Microsoft.ApplicationInsights.WindowsServer
+{
+    /// <summary>
+    /// A telemetry initializer that will gather configuration information.
+    /// </summary>
+    public class ConfigurationTelemetryInitializer : ITelemetryInitializer
```

Comment:
"Configuration" is an overloaded term. What configuration is this? Perhaps it should be called `CloudRoleInitializer` or something along those lines.

Reply:
Naming is one of the hardest tasks in software development. `CloudRoleInitializer`? Since it can also initialize the role instance.